### ISSUE TITLE
Rename Authentication Module to Controller Module 

### ISSUE THREAD

**Issue by seantking** [cosmos/ibc-go#1948]
## Summary

Rename all instances of  `authentication module` to `controller module` for ICA. This likely needs to be done for some godoc + developer documentation. 
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by colin-axner** [comment]
Could you open an issue on the spec repo as well?

**Comment by seantking** [comment]
> Could you open an issue on the spec repo as well? 

I don't think we reference authentication modules in the spec. I'll have a look. 

> Also, what is the preferred way to reference the controller module within this repo when compared against a controller module

I think we can just refer to our controller module as the "base" controller module. wdyt?

**Comment by crodriguezvega** [comment]
Consider opening smaller issues if there are too many things to change.
